src/psyclone/psyGen.py

```diff
@@ -3174,6 +3202,32 @@ def __str__(self):
         result += "EndLoop"
         return result
 
+    def reference_accesses(self, var_accesses):
+        '''Get all variable access information. It combines the data from
+        the loop bounds (start, stop, end step), as well as the loop body.
+        The loop variable is marked as READWRITE, start, stop, step as READ.
+        :param var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+        '''
+
+        # It is important to first add the WRITE access, since this way
+        # the dependency analysis for declaring openmp private variables
+        # will automatically declare the loop variables to be private
+        # (write access before read)
+        var_accesses.add_access(self.variable_name, AccessType.WRITE, self)
+        var_accesses.add_access(self.variable_name, AccessType.READ, self)
+        # TODO: self._start/stop/step are not defined at this stage (at least
```

Comment:
Please include an Issue number here. Also, the loop bounds might be arithmetic expressions (e.g. `jpi+1`). I guess we're waiting on the proper loop support that  @sergisiso is implementing in #400.